build-support/bin/packages.py

```diff
@@ -231,11 +343,50 @@ def build_pants_wheels() -> None:
                     f"expecting only one wheel: {sorted(wheel.name for wheel in found_wheels)}."
                 )
             for wheel in found_wheels:
-                if not (destination / wheel.name).exists():
+                if not (dest / wheel.name).exists():
                     # We use `copy2` to preserve metadata.
-                    shutil.copy2(wheel, destination)
+                    shutil.copy2(wheel, dest)
 
-    green(f"Wrote Pants wheels to {destination}.")
+    green(f"Wrote Pants wheels to {dest}.")
+
+    banner(f"Validating Pants wheels for {CONSTANTS.python_version}.")
+    create_twine_venv()
+    subprocess.run([CONSTANTS.twine_venv_dir / "bin/twine", "check", dest / "*.whl"], check=True)
+    green(f"Validated Pants wheels for {CONSTANTS.python_version}.")
```

Comment:
This used to run for `release.sh -t`, which was incorrect to do because it always inspects the `dist/` folder, whereas `-t` is only meant to test the live stable release on PyPI and we can make no assumptions that there are local wheels in `dist/`.

Now, it only runs when building wheels, which happens when building a PEX and when doing a dry release.